crates/containerd-shim-wasmedge/src/instance.rs

```diff
@@ -136,29 +131,6 @@ impl LibcontainerInstance for Wasi {
     }
 }
 
-impl EngineGetter for Wasi {
-    type Engine = Vm;
-    fn new_engine() -> Result<Vm, Error> {
-        PluginManager::load(None).unwrap();
-        let mut host_options = HostRegistrationConfigOptions::default();
-        host_options = host_options.wasi(true);
-        #[cfg(all(target_os = "linux", feature = "wasi_nn", target_arch = "x86_64"))]
-        {
-            host_options = host_options.wasi_nn(true);
-        }
-        let config = ConfigBuilder::new(CommonConfigOptions::default())
-            .with_host_registration_config(host_options)
-            .build()
-            .map_err(anyhow::Error::msg)?;
-        let vm = VmBuilder::new()
-            .with_config(config)
-            .build()
-            .map_err(anyhow::Error::msg)?;
-
-        Ok(vm)
-    }
-}
```

Comment:
This is a red herring.
It's actually never used, the `WasmEdgeExecutor` builds its own `Vm` from scratch.

Reply:
Oh I guess this answers my questions

Reply:
Should we build `Vm` in the instance and then passes it to Executor so that we enable sharing.

Reply:
Maybe.
I didn't share in this PR because the code here (removed) and the one in the executor were different, and wasn't sure the vm in the executor was shareable (it does things like installing plugins).
@CaptainVincent , do you have any input?

Reply:
From my memory, it was used before the refactor into the container/executor architecture, but the current structure indeed always creates a new one. The aspects of sharing and reusing also involve the implementation's support for wasm runtime context. I thought enabling instance sharing only at the shim level might be questionable. I think there shouldn't be an issue removing it at this point.